### PR TITLE
PoC Enabling the Base Media File Format support with Exiv2 0.27.4

### PR DESCRIPTION
**Pull request by 1div0**
Extended arrays so compilation finishes even with the warnings treated as errors.
Fix for #8339

### CODE REVIEW

**Comment by 1div0** [comment]
https://github.com/darktable-org/darktable/pull/8340/commits/51ee430e8b1563b3c3a159ca1cf27ae12634497b enables support of BMFF such as AVIF, HEIF and CR3 in upcoming Exiv2.
Snímka obrazovky z 2021-02-26 20-46-26

**Comment by clanmills** [comment]
Peter's there a macro is exiv2 to verify the version of the library in use.  I think you use it something like this.

```cpp
#include <exiv2/exiv2.hpp>
...
#if EXIV2_TEST_VERSION(0,27,4) 
   Exiv2::enableBMFF();
#endif
```

I've never used it of course.  It means that your code will compile with earlier versions of the library.

**Comment by Nilvus** [comment]
@1div0: could you please add "fix for #8339" in your first post content of this PR please? This will link #8339 issue to be closed automatically if this is merged. It would be good to update title to a title more explicit about what this PR does. Thanks!

File: src/iop/channelmixerrgb.c [hunk 1]
```diff
@@ -1333,7 +1333,7 @@ static const extraction_result_t _extract_patches(const float *const restrict in
     for(size_t c = 0; c < 3; c++) patches[k * 4 + c] /= (float)num_elem;
 
     // Convert to XYZ
-    float XYZ[3];
+    float XYZ[4];
```
Comment (ralfbrown) [inline src/iop/channelmixerrgb.c:1336]: Better to use `float DT_ALIGNED_PIXEL XYZ[4];`
as `dot_product` also declares its parameters aligned.

**Comment by johnny-bit** [comment]
Dunno if mixing a fix for compiler problem and upcoming exiv2 support in one PR is good practice, but I really like the BMFF support comming :)

**Comment by 1div0** [comment]
Indeed, I wanted to finish PoC quickly.

File: src/common/exif.cc [hunk 2]
```diff
@@ -4242,6 +4242,10 @@ void dt_exif_init()
   // preface the exiv2 messages with "[exiv2] "
   Exiv2::LogMsg::setHandler(&dt_exif_log_handler);
 
+#if EXIV2_TEST_VERSION(0,27,4)
```
Comment (kmilos) [inline src/common/exif.cc:4245]: Should be replaced w/

#ifdef EXV_ENABLE_BMFF

**Comment by mirecta** [comment]
Hhy this is inactive ? is there any problem ?

**Comment by johnny-bit** [comment]
@mirecta - it seems that `channelmixerrgb.c` part should be removed (fixes are done in tree now) and `EXV_ENABLE_BMFF` should be tested, but other that that I think it's pretty much ready.

**Comment by mirecta** [comment]
if this will be done does it meaning that Darktable will support CR3 files ?

**Comment by cytrinox** [comment]
> if this will be done does it meaning that Darktable will support CR3 files ?

No, this enables reading of EXIF metadata from CR3. The raw decoder (for the pixels) in rawspeed for CR3 is still missing and is afaik the biggest part.

**Comment by TurboGit** [comment]
Work done in #8751.

**Pull request closed without merging**